apis/lambda/v1alpha1/zz_types.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by ack-generate. DO NOT EDIT.
+
+package v1alpha1
+
+type AccountLimit struct {
+	CodeSizeUnzipped *int64 `json:"codeSizeUnzipped,omitempty"`
+
+	CodeSizeZipped *int64 `json:"codeSizeZipped,omitempty"`
+
+	TotalCodeSize *int64 `json:"totalCodeSize,omitempty"`
+}
+
+type AccountUsage struct {
+	FunctionCount *int64 `json:"functionCount,omitempty"`
+
+	TotalCodeSize *int64 `json:"totalCodeSize,omitempty"`
+}
+
+type AllowedPublishers struct {
+	SigningProfileVersionARNs []*string `json:"signingProfileVersionARNs,omitempty"`
+}
+
+type CodeSigningConfig_SDK struct {
+	// List of signing profiles that can sign a code package.
+	AllowedPublishers *AllowedPublishers `json:"allowedPublishers,omitempty"`
+
+	CodeSigningConfigARN *string `json:"codeSigningConfigARN,omitempty"`
+
+	CodeSigningConfigID *string `json:"codeSigningConfigID,omitempty"`
+	// Code signing configuration policies specifies the validation failure action
+	// for signature mismatch or expiry.
+	CodeSigningPolicies *CodeSigningPolicies `json:"codeSigningPolicies,omitempty"`
+
+	Description *string `json:"description,omitempty"`
+
+	LastModified *string `json:"lastModified,omitempty"`
+}
+
+type CodeSigningPolicies struct {
+	UntrustedArtifactOnDeployment *string `json:"untrustedArtifactOnDeployment,omitempty"`
+}
+
+type DeadLetterConfig struct {
+	TargetARN *string `json:"targetARN,omitempty"`
+}
+
+type Environment struct {
+	Variables map[string]*string `json:"variables,omitempty"`
+}
+
+type EnvironmentError struct {
+	ErrorCode *string `json:"errorCode,omitempty"`
+
+	Message *string `json:"message,omitempty"`
+}
+
+type EnvironmentResponse struct {
+	// Error messages for environment variables that couldn't be applied.
+	Error *EnvironmentError `json:"error,omitempty"`
+
+	Variables map[string]*string `json:"variables,omitempty"`
+}
+
+type FileSystemConfig struct {
+	ARN *string `json:"arn,omitempty"`
+
+	LocalMountPath *string `json:"localMountPath,omitempty"`
+}
+
+type FunctionCode struct {
+	ImageURI *string `json:"imageURI,omitempty"`
+
+	S3Bucket *string `json:"s3Bucket,omitempty"`
```

Comment:
Seems like this field could have referencer to `Bucket` object.

Reply:
Unfortunately the `FunctionCodeLocation` property returned as part of `GetFunctionOutput` does not map in any way to this `FunctionCode` object.   The properties on this object are used when creating the lambda function and can't be retrieved, although you can make an update request to change them.
https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#FunctionCodeLocation